### PR TITLE
fix: Users removed from the team are displayed as 'Default'

### PR DESCRIPTION
**Pull request by makingthematrix**
fixes: https://github.com/wireapp/android-project/issues/285

This is a temporary solution discussed with Design. In the future we will come up with something else, but now we just want to be consistent with what iOS uses.

Until now, when a user deleted their account, we simply marked her or his data  as 'deleted' in the databases of other users, connected to that one. But if it was a team user, we actually deleted the data in SE, while still expecting in UI to have it.
The SE part of this fix makes removed team users also marked as 'deleted'.
The UI part changes how deleted users' names are displayed: to 'Default'.

SE part: https://github.com/wireapp/wire-android-sync-engine/pull/435

#### APK
[Download build #11826](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11826/artifact/build/artifact/wire-dev-PR1824-11826.apk)
[Download build #11827](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11827/artifact/build/artifact/wire-dev-PR1824-11827.apk)
[Download build #11878](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11878/artifact/build/artifact/wire-dev-PR1824-11878.apk)
[Download build #11882](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11882/artifact/build/artifact/wire-dev-PR1824-11882.apk)
[Download build #11883](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11883/artifact/build/artifact/wire-dev-PR1824-11883.apk)
[Download build #11889](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11889/artifact/build/artifact/wire-dev-PR1824-11889.apk)
[Download build #11988](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11988/artifact/build/artifact/wire-dev-PR1824-11988.apk)
[Download build #12008](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12008/artifact/build/artifact/wire-dev-PR1824-12008.apk)
[Download build #12009](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12009/artifact/build/artifact/wire-dev-PR1824-12009.apk)
[Download build #12088](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12088/artifact/build/artifact/wire-dev-PR1824-12088.apk)
[Download build #12089](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12089/artifact/build/artifact/wire-dev-PR1824-12089.apk)
[Download build #12117](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12117/artifact/build/artifact/wire-dev-PR1824-12117.apk)
[Download build #12171](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12171/artifact/build/artifact/wire-dev-PR1824-12171.apk)
[Download build #12213](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12213/artifact/build/artifact/wire-dev-PR1824-12213.apk)
[Download build #12229](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12229/artifact/build/artifact/wire-dev-PR1824-12229.apk)

### CODE REVIEW

**Comment by red-avtovo** [comment]
Bug 1:
After member was removed from the team, his avatar and name just disappears

**Comment by red-avtovo** [comment]
screenshot_20181018-111648

**Comment by red-avtovo** [comment]
Bug 2
There is no system message "<User> was removed from the team"

**Comment by red-avtovo** [comment]
Bug 3
Avatar of deleted user is not reset to default

**Comment by red-avtovo** [comment]
screenshot_20181018-163006

**Comment by makingthematrix** [comment]
I fixed setting the avatar to default, but we don't currently have a system message for "Member was removed from the team" and I think we also lack functionality for it in SE. I would prefer to create a ticket for it and do it in another PR.

**Comment by red-avtovo** [comment]
Test

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -834,6 +834,9 @@
     <string name="all_archived__header">Everything archived</string>
     <string name="no_conversation_in_list__message">Start a conversation or create a group.</string>
 
+    <string name="user_deleted_username">Default</string>
+    <string name="user_deleted_initial">D</string>
```
Comment (red-avtovo) [inline app/src/main/res/values/strings.xml]: to Dev reviewer:
Should we really put hardcoded literal for Default user or we can reuse generic way of initials expression

**Comment by red-avtovo** [comment]
Bug 5:
Deleted user is still visible is search results. It allows to try to create a conversation with him and receive an error. Also it is possible to add this person back to conversation and receive even worse error message

**Comment by red-avtovo** [comment]
screenshot_20181030-110956

**Comment by red-avtovo** [comment]
I will stop testing this PR unless Devs will finish their review

**Comment by makingthematrix** [comment]
I removed deleted users form the search results, changed the initials, so that they're parsed from the default username instead of having their own string. Also when the user goest to a conversation with a deleted user, it is now impossible to access the participant's details.

**Comment by makingthematrix** [comment]
This PR was rebased on top of `public_logging`. Please note that for it to work correctly, the SE part of `public_logging` has to include the SE part of `default_username` (already in `develop`).

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1009,6 +1011,8 @@
     <string name="guest_indicator">GUEST</string>
     <string name="participant_tab_guest_indicator_label">Guest</string>
 
+    <string name="conversation_list__def_conv_name">Default</string>
```
[the remark this answers is not in the document]
Comment (makingthematrix) [reply app/src/main/res/values/strings.xml]: done

**Comment by red-avtovo** [comment]
-> Bug 1

**Comment by makingthematrix** [comment]
build again

**Comment by makingthematrix** [comment]
build again

**Comment by red-avtovo** [comment]
I can't login in the client

**Comment by red-avtovo** [comment]
screenshot_20190107-174736

Button on the bottom is not clickable

**Pull request merged**